docs/model-dev-guide/apis-howto/deepspeed/autotuning.rst

```diff
@@ -0,0 +1,303 @@
+.. _deepspeed-autotuning:
+
+################################
+ DeepSpeed Autotune: User Guide
+################################
+
+.. meta::
+   :description: This user guide demonstrates how to optimize DeepSpeed parameter in order to take full advantage of the user's hardware and model.
+
+Getting the most out of DeepSpeed (DS) requires aligning the many DS parameters with the specific
+properties of your hardware and model. Determined AI's DeepSpeed Autotune (``dsat``) helps to
+optimize these settings through an easy-to-use API with very few changes required in user-code, as
+we describe in the remainder of this user guide. ``dsat`` can be used with
+:class:`~determined.pytorch.deepspeed.DeepSpeedTrial`, :ref:`Core API <core-getting-started>`, and
+`HuggingFace Trainer <https://huggingface.co/docs/transformers/main_classes/trainer>`__.
+
+**************
+ How it Works
+**************
+
+You do not need to create a special configuration file to use ``dsat``. Assuming you have DeepSpeed
+code which already functions, autotuning is as easy as inserting one or two helper functions into
+your code and modifying the launch command.
+
+For instance, let us imagine that your present directory contains DeepSpeed code and a corresponding
+``single`` trial experiment configuration file ``deepspeed.yaml``. Then, after inserting a line or
+two of ``dsat``-specific code per the instructions in the following sections, launching the ``dsat``
+experiments is as easy as replacing the usual experiment-launching command
+
+.. code::
+
+   det experiment create deepspeed.yaml .
+
+with
+
+.. code::
+
+   python3 -m determined.pytorch.dsat asha deepspeed.yaml .
+
+The above uses Determined AI's DeepSpeed Autotune with the ``asha`` algorithm, one of three
+available search methods:
+
+-  ``asha``: Adaptively searches over randomly selected DeepSpeed configurations, allocating more
+   compute resources to well-performing configurations. See :ref:`this introduction to ASHA
+   <topic-guides_hp-tuning-det_adaptive-asha>` for more details.
+
+-  ``binary``: Performs a simple binary search over the batch size for randomly-generated DS
+   configurations.
+
+-  ``random``: Conducts a search over random DeepSpeed configurations with an aggressive
+   early-stopping criteria based on domain-knowledge of DeepSpeed and the search history.
+
+DeepSpeed Autotune is built on top of Custom Searcher (see :ref:`topic-guides_hp-tuning-det_custom`)
+which starts up two separate experiments:
+
+-  ``single`` Search Runner Experiment: This experiment coordinates and schedules the trials that
+   run the model code.
+-  ``custom`` Experiment: This experiment contains the trials referenced above whose results are
+   reported back to the search runner.
+
+Initially, a one-step profiling trial is created to gather information regarding the model and
```

Comment:
I feel like the wording of "one-step" might be confusing. Simply removing it sounds better to me. "Initially, a profiling trial is created [...]"